core/drand_daemon_control.go

```diff
@@ -0,0 +1,177 @@
+package core
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/drand/drand/key"
+
+	"github.com/drand/drand/common/scheme"
+
+	"github.com/drand/drand/protobuf/common"
+	"github.com/drand/drand/protobuf/drand"
+)
+
+// InitDKG take a InitDKGPacket, extracts the informations needed and wait for
+// the DKG protocol to finish. If the request specifies this node is a leader,
+// it starts the DKG protocol.
+func (dd *DrandDaemon) InitDKG(c context.Context, in *drand.InitDKGPacket) (*drand.GroupPacket, error) {
+	bp, beaconID, err := dd.getBeaconProcess(in.GetMetadata())
+	if err != nil {
+		store, isStoreLoaded := dd.initialStores[beaconID]
+		if !isStoreLoaded {
+			dd.log.Infow("", "beacon_id", beaconID, "init_dkg", "loading store from disk")
+
+			newStore := key.NewFileStore(dd.opts.ConfigFolderMB(), beaconID)
+			store = &newStore
+		}
+
+		dd.log.Infow("", "beacon_id", beaconID, "init_dkg", "instantiating a new beacon process")
+		bp, err = dd.AddNewBeaconProcess(beaconID, *store)
+		if err != nil {
+			return nil, fmt.Errorf("something went wrong try to initiate DKG. err: %s", err)
+		}
+	}
+
+	return bp.InitDKG(c, in)
+}
+
+// InitReshare receives information about the old and new group from which to
+// operate the resharing protocol.
+func (dd *DrandDaemon) InitReshare(ctx context.Context, in *drand.InitResharePacket) (*drand.GroupPacket, error) {
+	bp, beaconID, err := dd.getBeaconProcess(in.GetMetadata())
+	if err != nil {
+		store, isStoreLoaded := dd.initialStores[beaconID]
+		if !isStoreLoaded {
+			dd.log.Infow("", "beacon_id", beaconID, "init_reshare", "loading store from disk")
+
+			newStore := key.NewFileStore(dd.opts.ConfigFolderMB(), beaconID)
+			store = &newStore
+		}
+
+		dd.log.Infow("", "beacon_id", beaconID, "init_reshare", "instantiating a new beacon process")
+		bp, err = dd.AddNewBeaconProcess(beaconID, *store)
+		if err != nil {
+			return nil, fmt.Errorf("something went wrong try to initiate DKG")
+		}
+	}
+
+	return bp.InitReshare(ctx, in)
+}
+
+// PingPong simply responds with an empty packet, proving that this drand node
+// is up and alive.
+func (dd *DrandDaemon) PingPong(ctx context.Context, in *drand.Ping) (*drand.Pong, error) {
+	metadata := common.NewMetadata(dd.version.ToProto())
+	return &drand.Pong{Metadata: metadata}, nil
+}
+
+// Status responds with the actual status of drand process
+func (dd *DrandDaemon) Status(ctx context.Context, in *drand.StatusRequest) (*drand.StatusResponse, error) {
+	bp, _, err := dd.getBeaconProcess(in.GetMetadata())
+	if err != nil {
+		return nil, err
+	}
+
+	return bp.Status(ctx, in)
+}
+
+func (dd *DrandDaemon) ListSchemes(ctx context.Context, in *drand.ListSchemesRequest) (*drand.ListSchemesResponse, error) {
```

Comment:
Actually - do we need this method ??
Isn't `Version()` sufficient / better ? A Given versions tells what schemes do we support normally no ? 
Not saying we should change that here but I'm raising the point that I dont see why we have this particular methods instead of the more generic version().

Reply:
Schemes are not attached to a specific app version AFAIK. So it wouldn't be correct to infer the supported schemes from a particular app version?